File: python/triangle/triangle.py
def equilateral(sides):
    if is_triangle(sides):
        return sides[0] == sides[1] == sides [2]
    return False

def isosceles(sides):
    if is_triangle(sides):
        return sides[0] == sides[1] or sides[1] == sides[2] or sides[0] == sides[2]
    return False

def scalene(sides):
    if is_triangle(sides) and not isosceles(sides) and not equilateral(sides):
        return True
    return False

def is_triangle(sides):
    if sides[0] <=0 or sides[1] <=0 or sides[2] <= 0 or \
    sides[0] + sides[1] < sides[2] or sides[1] + sides[2] < sides[0] or sides[0] + sides[2] < sides[1]:
        return False
    return True

File: Exercism_scripts/go/triangle/triangle.go
//package triangle identify the type of traingles
package triangle

import (
	"math"
)

type Kind int

const (
	NaT Kind = iota // not a triangle
	Equ             // equilateral
	Iso             // isosceles
	Sca             // scalene
)

// KindFromSides identify the traingle type
func KindFromSides(a, b, c float64) Kind {

	if a <= 0 || b <= 0 || c <= 0 || math.IsNaN(a) || math.IsNaN(b) || math.IsNaN(c) || math.IsInf(a, 0) || math.IsInf(b, 0) || math.IsInf(c, 0) {
		return NaT
	}
	if (a > b+c) || (b > a+c) || (c > a+b) {
		return NaT
	}
	if (a == b) && (b == c) {
		return Equ
	}
	if (a == b) || (b == c) || (a == c) {
		return Iso
	}
	return Sca
}
